File: src/basics/video.py
import cv2 as cv


vid_capture = cv.VideoCapture('../data/videos/Cars.mp4')

if (vid_capture.isOpened() == False):
    print("Error reading the video file.")
else:
    #Get the frame rate
    fps = vid_capture.get(5)
    print("Frames per Second: ", fps, 'FPS')

    #Get frame count.
    frame_count = vid_capture.get(7)
    print("Frame count: ", frame_count)

while(vid_capture.isOpened()):
    ret, frame = vid_capture.read() #Returns a tuple, first element is a bool, second one is a frame.
    if ret  == True:
        cv.imshow('Frame', frame)
        cv.putText(frame, text=f"Frames per second: {fps}", org=(7, 240), 
                   fontFace=cv.FONT_HERSHEY_DUPLEX, fontScale=3, color=(0, 255, 0))

        key = cv.waitKey(20)
        
        if key == ord('q'):
            break
    else:
        break

#Release video and capture object.
vid_capture.release()
cv.destroyAllWindows()

File: src/basics/cpp/video.cpp
//Include libraries.
#include <opencv2/opencv.hpp>
#include <iostream>

//Namespace to nullify the use of cv::function sysntax.
using namespace std;
using namespace cv;

int main()
{
    //Initialize vid cap object.
    VideoCapture vid_capture("Images/cars.mp4");

    if (!vid_capture.isOpened())
    {
        cout << "Error opening video stream or file" << endl;
    }
    else
    {
        //Obtainn count and fps.
        int fps = vid_capture.get(5);
        cout << "Frames per Second: " << fps;         

        int frame_count = vid_capture.get(7);
        cout << "Frame count " << frame_count;
    }

    while(vid_capture.isOpened())
    {
        Mat frame;
        
        bool isSuccess = vid_capture.read(frame);

        if(isSuccess == true)
        {
            imshow("Frame", frame);
        }
        if(isSuccess == false)
        {
            cout << "Video is disconnected " << endl;
            break; 
        }

        int key = cv.waitKey(20);
        if(key == 'q')
        {
            cout << "q key is pressed, closing the video." << endl;
            break;
        }
    }
    vid_capture.release();
    cv.destroyAllWindow();
    return 0;
}
